src/components/structures/RoomView.js

```diff
@@ -180,6 +177,11 @@ module.exports = React.createClass({
         roomProm.then((room) => {
             this._calculatePeekRules(room);
             return this._initTimeline(this.props);
+        }).catch(() => {
```

Comment:
We'd presumably want to show the spinner until we've tried to peek though, so we'll have to set something when the peek request fails. Probably better to check the actual error if that's the concern.